src/store/modules/data/checkout.js

```diff
@@ -27,20 +26,39 @@ export const fetchCheckoutDataLoading = createAction(FETCH_CHECKOUT_DATA_LOADING
 // Thunks
 // ------------------------------------
 
-export function getCheckout () {
+export function getCheckout (poprMultihash) {
+  console.log(poprMultihash)
   return async dispatch => {
     dispatch(fetchCheckoutDataLoading(true))
+    let popr = null
     try {
-      // TODO: replace fixture data with stuff from PoPR
-      // TODO: handle error in UI
-      const vendorId = process.env.REACT_APP_VENDOR_ID || 'Qmtest'
-      const url = process.env.REACT_APP_MARKETPLACE_URL || 'http://localhost:4000'
-      const popr = await requestPopr(url, vendorId, {
-        // TODO: fix the placeholder price
-        amount: checkoutData.price * 10000,
-        currency_symbol: 'tBTC'
-      })
-      dispatch(fetchCheckoutDataSuccess(Object.assign({}, popr, checkoutData)))
+      if (poprMultihash) {
+        console.log('Reading PoPR at ' + poprMultihash)
+        const chluIpfs = await getChluIPFS()
+        // TODO: create API calls in Chlu to fetch PoPR instead of this manual thing
```

Comment:
It is just getting an object, right? How would the API call to fetch popr be different from a get? More specifically, how will the parameter be different?

Reply:
it will be

```javascript
popr = await chluIpfs.getPoPR(poprMultihash)
```

Instead of

```javascript
 const buffer = await chluIpfs.instance.ipfsUtils.get(poprMultihash)
 popr = await chluIpfs.instance.protobuf.PoPR.decode(buffer)
```

Not a big change but if we do the `chluIpfs.getPoPR` method then this code will survive internal changes in ChluIPFS.